resources/lib/cache.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: UTF-8 -*-
+#
+# Copyright (C) 2020, Team Kodi
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# pylint: disable=missing-docstring
+#
+# This is based on the metadata.tvmaze scrapper by Roman Miroshnychenko aka Roman V.M.
+
+"""Cache-related functionality"""
+
+import os
+import pickle
+import xbmcvfs
+
+from .utils import ADDON, logger
+
+try:
+    from typing import Optional, Text, Dict, Any  # pylint: disable=unused-import
+except ImportError:
+    pass
+
+
+def _get_cache_directory():  # pylint: disable=missing-docstring
+    # type: () -> Text
+    temp_dir = xbmcvfs.translatePath('special://temp')
+    cache_dir = os.path.join(temp_dir, 'scrapers', ADDON.getAddonInfo('id'))
+    if not xbmcvfs.exists(cache_dir):
+        xbmcvfs.mkdir(cache_dir)
+    logger.debug('the cache dir is ' + cache_dir)
+    return cache_dir
+
+
+CACHE_DIR = _get_cache_directory()  # type: Text
+
+
+def cache_show_info(show_id, data, lang, part):
```

Comment:
I don't see that you set/check cache TTL anywhere. Do you want to cache data forever?

Reply:
If you put files to the right location then kodi checks for any outdated files (and deletes them) each time before plugin started.

Reply:
Ok, I didn't know about that feature. Then I withdraw my comments about cache TTL setting. The caching mechanism itself still needs to be reworked, pickling is too unreliable.